main/src/library/MutDisjointSet.flix

```diff
@@ -0,0 +1,147 @@
+/*
+ *  Copyright 2023 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+pub enum MutDisjointSet[t: Type, r: Eff]({
+    rc = Region[r],
+    forest = MutMap[t, MutDisjointSet.Node[t, r], r]
+})
+
+mod MutDisjointSet {
+
+    pub enum Node[t: Type, r: Eff]({
+        rc = Region[r],
+        value = t,
+        parent = Ref[Option[Node[t, r]], r],
+        rank = Ref[Int32, r]
+    })
+
+    instance Eq[Node[t, r]] with Eq[t] {
```

Comment:
I am worried about this instance. I would remove it.

I think it is only used in one place, and there you might want to be more explicit anyway.